### PR TITLE
fixes #105: warn on unquoted keywords

### LINKED ISSUE #105: Warn about unquoted keyword-source values in ancestor-source-list
Warn about unquoted `keyword-source` values in `ancestor-source-list` as we do for `source-expression`s.

Example: `frame-ancestors self;`

### PR DESCRIPTION
**Pull request by shekyan**

### CODE REVIEW

**Comment by shekyan** [comment]
BTW, I am open to add `'unsafe-redirect'` (deprecated one) to the list of keywords we warn about.

File: src/main/java/com/shapesecurity/salvation/Parser.java [hunk 1]
```diff
@@ -505,6 +498,7 @@ private void enforceMissingDirectiveValue(@Nonnull Token directiveNameToken) thr
         if (token.value.equalsIgnoreCase("'self'")) {
             return KeywordSource.Self;
         }
+        checkForUnquotedKeyword(token);
         if (token.value.matches("^" + Constants.schemePart + ":$")) {
```
Comment (michaelficarra) [inline src/main/java/com/shapesecurity/salvation/Parser.java]: From here to the end of the function is the same as 438-460. Is there any way we can dedupe this as well? And we can probably even do the same for `'none'` and `'self'` above. If you want, I can take a crack at it.

Comment (shekyan) [inline src/main/java/com/shapesecurity/salvation/Parser.java]: I'll do it

**Comment by shekyan** [comment]
@michaelficarra check it out

**Comment by shekyan** [comment]
@michaelficarra let push this in.

**Comment by michaelficarra** [comment]
I think I can make it cleaner. I will take a look as soon as I get the chance.

**Pull request merged**